### PR TITLE
GG-34281 Get jobs info by taskSessionId

### PR DESCRIPTION
**Pull request by tkalkirill**
https://ggsystems.atlassian.net/browse/GG-34281

### CODE REVIEW

File: modules/core/src/test/java/org/apache/ignite/internal/processors/compute/ComputeJobStatusTest.java [hunk 1]
```diff
@@ -0,0 +1,439 @@
...
+import static org.apache.ignite.internal.processors.job.ComputeJobStatusEnum.RUNNING;
+import static org.apache.ignite.internal.processors.job.ComputeJobStatusEnum.SUSPENDED;
+import static org.apache.ignite.testframework.GridTestUtils.getFieldValue;
+
+/**
+ * Class for testing {@link GridTaskProcessor#jobStatuses} and {@link GridJobProcessor#jobStatuses}.
+ */
+public class ComputeJobStatusTest extends GridCommonAbstractTest {
+    /** Coordinator. */
+    private static IgniteEx N0;
```
Comment (ibessonov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/compute/ComputeJobStatusTest.java:75]: Very strange naming, why not `node0` and `node1`? These are not final

File: modules/core/src/test/java/org/apache/ignite/internal/processors/compute/ComputeJobStatusTest.java [hunk 2]
```diff
@@ -0,0 +1,439 @@
...
+        checkTaskJobStatuses(sesId, RUNNING, null);
+        checkJobJobStatuses(sesId, RUNNING, RUNNING);
+
+        switch (exp) {
+            case FINISHED:
+                waitJob0.waitFut.onDone();
+                break;
+
+            case FAILED:
+                waitJob0.waitFut.onDone(new Exception("form test"));
```
Comment (ibessonov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/compute/ComputeJobStatusTest.java:218]: Exception message has a typo

File: modules/core/src/test/java/org/apache/ignite/internal/processors/compute/ComputeJobStatusTest.java [hunk 3]
```diff
@@ -0,0 +1,439 @@
...
+import static org.apache.ignite.internal.processors.job.ComputeJobStatusEnum.FINISHED;
+import static org.apache.ignite.internal.processors.job.ComputeJobStatusEnum.QUEUED;
+import static org.apache.ignite.internal.processors.job.ComputeJobStatusEnum.RUNNING;
+import static org.apache.ignite.internal.processors.job.ComputeJobStatusEnum.SUSPENDED;
+import static org.apache.ignite.testframework.GridTestUtils.getFieldValue;
+
+/**
+ * Class for testing {@link GridTaskProcessor#jobStatuses} and {@link GridJobProcessor#jobStatuses}.
+ */
+public class ComputeJobStatusTest extends GridCommonAbstractTest {
```
Comment (ibessonov) [inline modules/core/src/test/java/org/apache/ignite/internal/processors/compute/ComputeJobStatusTest.java:73]: I think you should add more comments throughout the test, it's quite complex

Comment (tkalkirill) [reply modules/core/src/test/java/org/apache/ignite/internal/processors/compute/ComputeJobStatusTest.java:73]: Added some comments.

**Review by ibessonov — APPROVED** [review]

File: modules/core/src/main/java/org/apache/ignite/internal/processors/job/ComputeJobStatusEnum.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * Compute job status.
+ */
+public enum ComputeJobStatusEnum {
+    /** */
+    QUEUED,
+
+    /** */
+    RUNNING,
```
Comment (ingvard) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/job/ComputeJobStatusEnum.java:27]: Empty java doc for all enums declarations.

**Review by ingvard — APPROVED** [review]

**Pull request merged**